src/vsphere_cpi/lib/cloud/vsphere/resources/cluster_provider.rb

```diff
@@ -25,6 +26,35 @@ def find(name, config)
         )
       end
 
+      # Delete VM Groups if they are empty
+      # @param [Vim::ClusterComputeResource] cluster
+      # @param [String[]] vm_group_names - List of VM Groups
+      def delete_vm_groups(cluster, vm_group_names)
```

Comment:
I was going to do that as a next step as the original implementation was to create DRS rule (vm/host) and adding vm to vm_group was just a part of it. We decided to not create rule so I made minimum changes,  so that when we add the rule it will be easy. 
Let me think about it but I agree with your point and went back and forth with how to do it